authority/provisioner/sign_options.go

```diff
@@ -197,7 +184,34 @@ func (v emailAddressesValidator) Valid(req *x509.CertificateRequest) error {
 	return nil
 }
 
-// validityValidator validates the certificate temporal validity settings.
+// profileProvCredDuration adjusts the duration to the
+// min(default, remaining provisioning credential duration.
+type profileProvCredDuration struct {
+	def time.Duration
+	rem time.Duration
+}
+
+func (v profileProvCredDuration) Option(so Options) x509util.WithOption {
+	if v.rem < v.def {
+		v.def = v.rem
+	}
+	return profileDefaultDuration(v.def).Option(so)
```

Comment:
If the user was the one who set the duration you're not checking that is lower than v.rem and returning an error.